src/components/Card/TodosCard/TodosCard.tsx

```diff
@@ -0,0 +1,272 @@
+          })
+        );
+      } catch (error) {
+        fireSwalError("The todo couldn't be updated");
+      }
+    },
+    [remaindersWorker]
+  );
+
+  const handleFilterChange = useCallback(
+    (event: React.MouseEvent<HTMLElement>, newFilter: string) => {
+      setTodosFilter(newFilter);
+    },
+    []
+  );
+
+  // TODO: the name doesn't imply boolean result, but a filter action - rename
+  // ANSWER: I'm not sure which is better "toShowTodo" or "isTodoFiltered"
+  const toShowTodo = (todoToFilter: Todo): boolean => {
+    return todosFilter === Filter.ALL
+      ? true
+      : (todoToFilter.isCompleted ? Filter.COMPLETED : Filter.ACTIVE) ===
+          todosFilter;
+  };
+
+  const openTimeModal = useCallback((todo: Todo) => {
+    setTimeModalState(true);
+    setTodoToUpdate({ ...todo });
+  }, []);
+
+  return (
+    <React.Fragment>
+      <Box m="auto" width="50%">
+        <Card>
+          <TodoInput onSubmit={handleTodoSubmit} />
+          <List>
+            {S.pipe([
+              S.filter(toShowTodo),
+              S.map((todo: Todo) => (
+                // TODO: if you're not familiar with react reconcilation, then please read (https://reactjs.org/docs/reconciliation.html) + read about react diff alogirthm from the same source
+                // and explain why we need the `key` prop
+                // ANSWER: rendering a list isn't easy for react diff algorithm, if an element is added at the end of the list
+                // then it's easy, react diff algo will compare all elements and find out they are the same, but just one element was added to the end.
+                // When an element is added at the start of the list, react will have a hard time finding out that the rest of the elements are the same,
+                // and finding this out is pretty expensive. So a solution react came up with is binding a key to each element rendered from a list, that way
+                // react diff algo will be able to know if elements position just moved, or their state changed.
+                <TodoItem
+                  key={todo.id}
+                  todo={todo}
+                  onDelete={handleTodoDelete}
+                  onTodoUpdate={handleTodoUpdate}
+                  openTimeModal={openTimeModal}
+                />
+              )),
+            ])(todos)}
+          </List>
+          <CardFooter
+            todosFilter={todosFilter}
+            numberOfUncompletedTodos={useMemo(getNumberOfUncompletedTodos, [
+              todos,
+            ])}
+            handleFilterChange={handleFilterChange}
+            clearAllCompleted={clearAllCompleted}
+          />
+        </Card>
+      </Box>
+      {/* ANSWER: The pervious solution was wrong because at first it wasn't scalable, adding more modals means adding more props */}
+      {/* which in some point will be too much, second, every prop change will cause rerendering of all modals (although could be partially solved with React.memo). */}
```

Comment:
the solution with react memo is not a good practice, because when you use react memo it costs with performance.
so if you designed your code poorly, you should not cover it up with other solutions rather than redesign in early stages